zkevm-framework.nix

```diff
@@ -0,0 +1,59 @@
+{ lib,
+  stdenv,
+  src_repo,
+  ninja,
+  pkg-config,
+  cmake,
+  boost183,
+  # We'll use boost183 by default, but you can override it
+  boost_lib ? boost183,
+  gdb,
+  crypto3,
+  ethash,
+  intx,
+  sszpp,
+  valijson,
+  gtest,
+  evm-assigner,
+  enableDebugging,
+  enableDebug ? false,
+  runTests ? false,
+  }:
+let
+  inherit (lib) optional;
+in stdenv.mkDerivation rec {
+  name = "zkevm-framework";
+
+  src = src_repo;
+
+  nativeBuildInputs = [ cmake ninja pkg-config ] ++ (lib.optional (!stdenv.isDarwin) gdb);
+
+  # enableDebugging will keep debug symbols in boost
+  propagatedBuildInputs = [ (if enableDebug then (enableDebugging boost_lib) else boost_lib) ];
+
+  buildInputs = [crypto3 evm-assigner intx ethash sszpp valijson gtest];
+
+  cmakeFlags =
+  [
+      (if runTests then "-DBUILD_TESTS=TRUE" else "")
+      (if runTests then "-DCMAKE_ENABLE_TESTS=TRUE" else "")
+      (if runTests then "-DENABLE_TESTS=TRUE" else "")
```

Comment:
Why do we have so much flags to enable tests? I think it makes sense to create an issue at least, and unify all repositories to use the same flag. Currently it's very easy to forget about some flag and leave corresponding tests disabled.

Reply:
good catch, seems it was copy-past.
`ENABLE_TESTS` - enough